zmq_addon.hpp

```diff
@@ -154,7 +155,122 @@ send_multipart(socket_ref s, Range &&msgs, send_flags flags = send_flags::none)
     return msg_count;
 }
 
+/* Encode a multipart message.
+
+   The range must be a ForwardRange of zmq::message_t.  A
+   zmq::multipart_t or STL container may be passed for encoding.
+
+   Returns: a zmq::message_t holding the encoded multipart data.
+
+   Throws: std::range_error is thrown if the size of any single part
+   can not fit in an unsigned 32 bit integer.
+
+   The encoding is compatible with that used by the CZMQ function
+   zmsg_encode().  Each part consists of a size followed by the data.
+   These are placed contiguously into the output message.  A part of
+   size less than 255 bytes will have a single byte size value.
+   Larger parts will have a five byte size value with the first byte
+   set to 0xFF and the remaining four bytes holding the size of the
+   part's data.
+*/
+template<class Range
+#ifndef ZMQ_CPP11_PARTIAL
+         ,
+         typename = typename std::enable_if<
+           detail::is_range<Range>::value
+           && (std::is_same<detail::range_value_t<Range>, message_t>::value
+               || detail::is_buffer<detail::range_value_t<Range>>::value)>::type
 #endif
+         >
+message_t encode(const Range &parts)
```

Comment:
Do you think `encode_multipart` or `multipart_encode` (and `decode_multipart` or `multipart_decode`) would be more descriptive?

Reply:
I do not have a strong opinion. but I lean to leaving this as-is or maybe with a different `_qualifier`.  

I'm thinking about future types which may overload `encode()` where the label `_multipart` is not meaningful.  Eg, an application defined  `encode(const MyMessage& mm)` would not necessarily have any "multipartish".

If we had a good name for this codec then `encode_<codecname>()` (etc) would maybe be a better way to distinguish this.